WEB_SERVICE/WEB/_CRITICAL/NOSQL_INJECTION/description.md

```diff
@@ -0,0 +1,95 @@
+NoSQL injection is a security vulnerability that occurs when attackers exploit weaknesses in applications using NoSQL databases. It involves injecting malicious or unexpected data into input fields, enabling attackers to manipulate queries and access unauthorized information.
+
+Similar to SQL injection attacks against SQL databases, NoSQL injection targets applications utilizing NoSQL databases like MongoDB, Cassandra, or others. Attackers craft input, such as special characters or payloads, to manipulate NoSQL database queries executed by the application.
+
+By injecting the crafted input, attackers aim to alter the logic of database queries, bypass authentication, or retrieve sensitive information stored in the database. NoSQL injection attacks can lead to unauthorized data access, data modification, or even the complete compromise of the database, compromising the confidentiality, integrity, and availability of the application and its data.
+
+
+
+
+### Examples
+
+=== Java
+  ```java
+    import com.mongodb.*;
+    import org.springframework.web.bind.annotation.*;
+    
+    @RestController
+    @RequestMapping("/login")
+    public class NoSQLInjectionController {
+    
+        @PostMapping
+        public void login(@RequestParam String userName, @RequestParam String password) {
+            MongoClientURI uri = new MongoClientURI("mongodb://localhost:27017");
+            MongoClient mongoClient = new MongoClient(uri);
+    
+            try {
+                DB database = mongoClient.getDB("test");
+                DBCollection collection = database.getCollection("users");
+    
+                BasicDBObject query = new BasicDBObject();
+                query.put("$where", "this.sharedWith == \"" + userName + "\" && this.password == \"" + password + "\"");
+    
+                DBCursor cursor = collection.find(query);
+    
+                while (cursor.hasNext()) {
+                    System.out.println(cursor.next());
+                }
+            } finally {
+                mongoClient.close();
+            }
+        }
+    }
+  ```
+
+=== Javascript
+
+  ```javascript
+    const MongoClient = require('mongodb').MongoClient;
+    const express = require('express');
+    const app = express();
+    
+    app.post('/login', async (req, res) => {
+        const username = req.body.username;
+        const password = req.body.password;
+    
+        const uri = 'mongodb://localhost:27017';
+        const client = new MongoClient(uri);
+    
+        try {
+            await client.connect();
+            const database = client.db('test');
+            const usersCollection = database.collection('users');
+    
+            // Vulnerable query susceptible to NoSQL Injection
```

Comment:
The same comment applies to the other examples.

Reply:
For php :  I've changed to be based on this : 
![image](https://github.com/Ostorlab/KB/assets/146742880/0b5f74c2-5f20-465a-8dca-9fb12aab6550)
For js : 
![image](https://github.com/Ostorlab/KB/assets/146742880/75228d77-c0a2-4e3e-af59-469fd9319c0a)